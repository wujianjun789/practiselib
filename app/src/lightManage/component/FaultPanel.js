/**
 * Created by a on 2017/8/28.
 */
import React, {Component} from 'react';
/**
 * Panel component
 * @param {String}      className   'panel theme,default:　panel-primary'
 * @param {String}      title       'panel title, default: '　'
 * @param {Any}         body        'panel body'
 * @param {Any}         footer      'panel footer,default:hidden'
 * @param {String}      text        'no body tips,default: '无相关数据!''
 * @param {Boolean}     closeBtn    'panel close,true: display, false: hidden, default: false'
 * @param {Func}        closeClick  'panel close button handler'
 *
 */
export default class FaultPanel extends Component {

  closeClick() {
    this.props.closeClick ? this.props.closeClick() : null;
  }

  render() {
    let clsName = 'panel ' + (this.props.className ? this.props.className : 'panel-primary');
    let style = this.props.style ? this.props.style : null;
    let props = this.props;
    let _props = {
      title: props.title ? props.title : '　',
      body: props.body ? props.body :
        <div className="row pull-center">{props.text ? props.text : 'No data'}</div>,
      footer: props.footer ? <div className="panel-footer clearfix">{props.footer}</div> : null,
      closeBtn: !!props.closeBtn,
    };

    return (
      <div className={clsName} style={style}>
        <div className="panel-heading">
          <h3 className="panel-title">{_props.title}</h3>
          {_props.closeBtn ? <button type="button" className="close" 
            onClick={() => this.closeClick()}><span>&times;</span></button> : null}
        </div>
        <div className="panel-body">
          {this.props.children ? this.props.children : _props.body }
        </div>
        { _props.footer }
      </div>
    );
  }
}