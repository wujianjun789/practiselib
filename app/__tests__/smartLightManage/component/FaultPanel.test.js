/**
 * Created by a on 2017/9/30.
 */
import React from 'react';
import {shallow} from 'enzyme'
import renderer from 'react-test-renderer';

import FaultPanel from '../../../src/smartLightManage/component/FaultPanel';

test('renders Scene', ()=>{
    const component = renderer.create(
        <FaultPanel />
    ).toJSON();

    expect(component).toMatchSnapshot();
});

test('renders Scene data', ()=>{
    const component = shallow(<FaultPanel />);
})