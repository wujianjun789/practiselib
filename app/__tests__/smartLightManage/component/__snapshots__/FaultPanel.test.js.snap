exports[`test renders Scene 1`] = `
<div
  className="panel panel-primary"
  style={null}>
  <div
    className="panel-heading">
    <h3
      className="panel-title">
      ã€€
    </h3>
  </div>
  <div
    className="panel-body">
    <div
      className="row pull-center">
      No data
    </div>
  </div>
</div>
`;
